ksql-execution/src/main/java/io/confluent/ksql/execution/codegen/CodeGenSpec.java

```diff
@@ -77,14 +83,20 @@ public void resolve(GenericRow row, Object[] parameters) {
     }
   }
 
+  public String getSchemaName(StructExpression structExpression) {
+    return structToCodeName.get(structExpression);
+  }
+
   static class Builder {
 
     private final ImmutableList.Builder<ArgumentSpec> argumentBuilder = ImmutableList.builder();
     private final Map<ColumnRef, String> columnRefToName = new HashMap<>();
     private final ImmutableListMultimap.Builder<FunctionName, String> functionNameBuilder =
         ImmutableListMultimap.builder();
+    private final Map<StructExpression, String> structToSchemaName = new HashMap<>();
 
     private int argumentCount = 0;
+    private int schemaCount = 0;
```

Comment:
```suggestion
    private int structConstructorCount = 0;
```

Reply:
The distinction between `struct` and `schema` is intentional in the code. The expression evaluator takes in as an argument the _schemas_ for the structs, not the struct expressions themselves. If you look at the code, it is used as `new Struct(schema0)...`. In this code, `schema0` is very much a schema.

It's just like how we have the method `addFunction` and not `addFunctionName`. The `functionName` is the key, but the `function` is what is actually being stored.

Resolving the other conversations that are about this to concentrate on this one.

Reply:
I'll name it `getStructSchemaName` and make those corresponding changes